laravel-app/app/Services/ProfileService.php

```diff
@@ -30,20 +32,49 @@ public function updateProfile(User $user, array $validatedData): Void
      */
     public function updateProfileImage(User $user, UpdateProfileImageRequest $request): void
     {
-        if ($request->hasFile('profile_image')) {
-            $file = $request->file('profile_image');
-            if ($file instanceof UploadedFile) {
-                $fileName = time() . '_' . $file->getClientOriginalName();
-
-                $file->storeAs('public/profile_images', $fileName);
-
-                $userMedia = new UserMedia([
-                    'user_id' => $user->id,
-                    'file_path' => 'storage/profile_images/' . $fileName,
-                    'file_name' => $fileName,
-                ]);
-                $userMedia->save();
+        DB::beginTransaction();
+
+        try {
+            if ($request->hasFile('profile_image')) {
+
+                UserMedia::where('user_id', $user->id)->delete();
+
+                $file = $request->file('profile_image');
+                if ($file instanceof UploadedFile) {
+                    $fileName = time() . '_' . $file->getClientOriginalName();
+
+                    $file->storeAs('public/profile_images', $fileName);
+
+                    $userMedia = new UserMedia([
+                        'user_id' => $user->id,
+                        'file_path' => 'storage/profile_images/' . $fileName,
+                        'file_name' => $fileName,
+                    ]);
+                    $userMedia->save();
+                }
             }
+
+            DB::commit();
+        } catch (Exception $e) {
+            DB::rollBack();
+        }
+    }
```

Comment:
Since you are using DB transaction. Much better if you use it to your controller instead of per service right? if ever na mag failed yung controller lahat ng saving I aabort niya.

Reply:
Ohh okay sir 🙇 I'll resolve it now 🙇